From akpm@linux-foundation.org  Tue May  4 13:02:40 2010
From: Roel Kluin <roel.kluin@gmail.com>
Date: Tue, 27 Apr 2010 14:16:33 -0700
Subject: serial: two branches the same in timbuart_set_mctrl()
To: greg@kroah.com
Cc: akpm@linux-foundation.org, roel.kluin@gmail.com, alan@lxorguk.ukuu.org.uk, richard.rojfors@pelagicore.com
Message-ID: <201004272116.o3RLGXku020373@imap1.linux-foundation.org>


From: Roel Kluin <roel.kluin@gmail.com>

CTS is a read only bit and we are to stop signal RTS if modem line
TIOCM_RTS is not set.

Thanks for suggestions by Richard Röjfors.

Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
Acked-by: Richard Röjfors <richard.rojfors@pelagicore.com>
Cc: Alan Cox <alan@lxorguk.ukuu.org.uk>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/serial/timbuart.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/serial/timbuart.c
+++ b/drivers/serial/timbuart.c
@@ -220,7 +220,7 @@ static void timbuart_set_mctrl(struct ua
 	if (mctrl & TIOCM_RTS)
 		iowrite8(TIMBUART_CTRL_RTS, port->membase + TIMBUART_CTRL);
 	else
-		iowrite8(TIMBUART_CTRL_RTS, port->membase + TIMBUART_CTRL);
+		iowrite8(0, port->membase + TIMBUART_CTRL);
 }
 
 static void timbuart_mctrl_check(struct uart_port *port, u32 isr, u32 *ier)
